src/meson.build

```diff
@@ -5,6 +5,11 @@
 #
 # Authors: Martin Belanger <Martin.Belanger@dell.com>
 #
+configure_file(
+    output: 'config.h',
+    configuration: conf
+)
+
 sources += [
```

Comment:
Hi @igaw 

We could directly add the ccan source files here and completely eliminate the need for `ccan/meson.build`.

For example:
```
sources = [
    '../ccan/ccan/list/list.c',
    '../ccan/ccan/str/debug.c',
    '../ccan/ccan/str/str.c',
    'nvme/cleanup.c',
    'nvme/fabrics.c',
    'nvme/filters.c',
    'nvme/ioctl.c',
    'nvme/linux.c',
    'nvme/log.c',
    'nvme/tree.c',
    'nvme/util.c',
]
```
I can submit a pull request if you'd like.

Reply:
I don't have any strong feelings on this. If the general mode is that we need to get rid of the ccan/meson.build, please go ahead.

Reply:
This is weird. The 3 ccan files that are added to `sources` are actually not needed at all. I cannot find any libnvme code that uses any of the functions defined in these 3 files. I just tried building without these files and I ran all the tests successfully. I will submit a pull request that cleans all of that up.